lib/index.js

```diff
@@ -137,40 +134,42 @@ function configure() {
   });
 }
 
+async function listenHttps() {
+  const logger = bedrock.loggers.get('app');
+  const port = bedrock.config.server.port;
+  return Promise.all(bedrock.config.server.bindAddr.map(async addr => {
+    logger.debug('starting HTTPS server on %s:%d', addr, port);
+    await bedrock.events.emit(
+      'bedrock-server.https.listen', {address: addr, port});
+    await new Promise(resolve => {
+      api.servers.https.listen(port, addr, () => resolve());
+    });
+    await bedrock.events.emit(
+      'bedrock-server.https.listening', {address: addr, port});
+  }));
+}
+
+async function listenHttp() {
+  const logger = bedrock.loggers.get('app');
+  const port = bedrock.config.server.httpPort;
+  return Promise.all(bedrock.config.server.bindAddr.map(async addr => {
+    logger.debug('starting HTTP server on %s:%d', addr, port);
+    await bedrock.events.emit(
+      'bedrock-server.http.listen', {address: addr, port});
+    await new Promise(resolve => {
+      api.servers.http.listen(port, addr, () => resolve());
+    });
+    await bedrock.events.emit(
+      'bedrock-server.http.listening', {address: addr, port});
+  }));
+}
+
 /**
  * Start listening on the configured ports.
- *
- * @param callback(err) called once the operation completes.
  */
-function listen(callback) {
-  async.auto({
-    listenHttps: callback => {
-      const logger = bedrock.loggers.get('app');
-      const port = bedrock.config.server.port;
-      async.each(bedrock.config.server.bindAddr, (addr, next) => {
-        logger.debug('starting HTTPS server on %s:%d', addr, port);
-        bedrock.events.emit(
-          'bedrock-server.https.listen', {address: addr, port});
-        api.servers.https.listen(port, addr, () => {
-          bedrock.events.emit(
-            'bedrock-server.https.listening',
-            {address: addr, port});
-          next();
-        });
-      }, callback);
-    },
-    listenHttp: callback => {
-      const logger = bedrock.loggers.get('app');
-      const port = bedrock.config.server.httpPort;
-      async.each(bedrock.config.server.bindAddr, (addr, next) => {
-        logger.debug('starting HTTP server on %s:%d', addr, port);
-        api.servers.http.listen(port, addr, () => next());
-      }, callback);
-    },
-    ready: ['listenHttps', 'listenHttp', (results, callback) => {
-      const logger = bedrock.loggers.get('app');
-      logger.info('started server on port ' + bedrock.config.server.port);
-      bedrock.events.emit('bedrock-server.ready', callback);
-    }]
-  }, err => callback(err));
+async function listen() {
+  await Promise.all([listenHttps(), listenHttp()]);
```

Comment:
Would it be easier to debug things if these were done using `await` so that things are always started and logged in the same order?  Sounds like something I would like if I were reviewing logs.

Reply:
It would be more deterministic with individual awaits, but in this case I was specifically trying to do all the listens in parallel to maybe reduce the startup latency.

Reply:
Leaving the parallel version in for next release for the potential performance issue.  If it's a problem in practice, we can serialize these.